source/extensions/outlier_detection_monitors/common/monitor_base_impl.cc

```diff
@@ -0,0 +1,81 @@
+#include "source/extensions/outlier_detection_monitors/common/monitor_base_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Outlier {
+
+bool HTTPCodesBucket::matches(
+    const TypedExtResult<Upstream::Outlier::ExtResultType::HTTP_CODE>& result) const {
+  // We should not get here with errors other then HTTP codes.
+  ASSERT(result.type() == ExtResultType::HTTP_CODE);
+  const HttpCode& http_code = dynamic_cast<const HttpCode&>(result);
+  return ((http_code.code() >= start_) && (http_code.code() <= end_));
+}
+
+bool LocalOriginEventsBucket::matches(
+    const TypedExtResult<Upstream::Outlier::ExtResultType::LOCAL_ORIGIN>& event) const {
+  // We should not get here with errors other then Local Origin events.
+  ASSERT(event.type() == ExtResultType::LOCAL_ORIGIN);
```

Comment:
Agree, but this is for unit tests only. This method should not be called without calling `matchType` method first. The framework which looks for matching buckets will reject that type because `matchType` will fail and this `matches` method will never be called for non-matching types.